phpBB/phpbb/module/module_manager.php

```diff
@@ -0,0 +1,570 @@
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\module;
+
+use phpbb\module\exception\module_class_not_defined_exception;
+use phpbb\module\exception\module_exception;
+use phpbb\module\exception\module_not_found_exception;
+
+class module_manager
+{
+	/**
+	 * @var \phpbb\cache\driver\driver_interface
+	 */
+	protected $cache;
+
+	/**
+	 * @var \phpbb\db\driver\driver_interface
+	 */
+	protected $db;
+
+	/**
+	 * @var \phpbb\extension\manager
+	 */
+	protected $extension_manager;
+
+	/**
+	 * @var string
+	 */
+	protected $modules_table;
+
+	/**
+	 * @var string
+	 */
+	protected $phpbb_root_path;
+
+	/**
+	 * @var string
+	 */
+	protected $php_ext;
+
+	/**
+	 * Constructor
+	 *
+	 * @param \phpbb\cache\driver\driver_interface	$cache				Cache driver
+	 * @param \phpbb\db\driver\driver_interface		$db					Database driver
+	 * @param \phpbb\extension\manager				$ext_manager		Extension manager
+	 * @param string								$modules_table		Module database table's name
+	 * @param string								$phpbb_root_path	Path to phpBB's root
+	 * @param string								$php_ext			Extension of PHP files
+	 */
+	public function __construct(\phpbb\cache\driver\driver_interface $cache, \phpbb\db\driver\driver_interface $db, \phpbb\extension\manager $ext_manager, $modules_table, $phpbb_root_path, $php_ext)
+	{
+		$this->cache				= $cache;
+		$this->db					= $db;
+		$this->extension_manager	= $ext_manager;
+		$this->modules_table		= $modules_table;
+		$this->phpbb_root_path		= $phpbb_root_path;
+		$this->php_ext				= $php_ext;
+	}
+
+	/**
+	 * Get row for specified module
+	 *
+	 * @param int		$module_id		ID of the module
+	 * @param string	$module_class	Class of the module (acp, ucp, mcp etc...)
+	 *
+	 * @return array	Array of data fetched from the database
+	 *
+	 * @throws \phpbb\module\exception\module_not_found_exception	When there is no module with $module_id
+	 */
+	public function get_module_row($module_id, $module_class)
+	{
+		$module_id = (int) $module_id;
+
+		$sql = 'SELECT *
+			FROM ' . $this->modules_table . "
+			WHERE module_class = '" . $this->db->sql_escape($module_class) . "'
+				AND module_id = $module_id";
+		$result = $this->db->sql_query($sql);
+		$row = $this->db->sql_fetchrow($result);
+		$this->db->sql_freeresult($result);
+
+		if (!$row)
+		{
+			throw new module_not_found_exception('NO_MODULE');
+		}
+
+		return $row;
+	}
+
+	/**
+	 * Get available module information from module files
+	 *
+	 * @param string	$module				ID of module
+	 * @param string	$module_class		Class of the module (acp, ucp, mcp etc...)
+	 * @param bool		$use_all_available	Use all available instead of just all
+	 *										enabled extensions
+	 *
+	 * @return array	Array with module information gathered from module info files.
+	 */
+	public function get_module_infos($module = '', $module_class = null, $use_all_available = false)
+	{
+		if ($module_class === null)
```

Comment:
That's what I tried to ask [here](https://github.com/phpbb/phpbb/pull/3552/files#r31227530). So I would agree.